File: Algorithms/Tree/Height of the tree/CPP/bst_height.cpp
//Finding Height of Binary Search Tree

#include<bits/stdc++.h>
using namespace std;

struct node
{
	int info;
	struct node *lchild;
	struct node *rchild;
};

//Insertion in binary search tree
struct node *insert_bs( struct node *root , int data)
{
	//Creation of node
	struct node *temp;
	temp = (struct node*)malloc(sizeof(struct node));
	temp->info = data;
	temp->rchild = NULL;
	temp->lchild = NULL;

    if(root == NULL)
    	return temp;
    struct node *parent = NULL , *curr = root;
    while(curr != NULL)
    {
      parent = curr;
      if(curr->info < data)
        curr = curr->rchild;
      else if(curr->info > data)
       curr = curr->lchild;  
    }
    if(parent->info < data)
    	parent->rchild = temp;
    else if(parent->info > data)
    	parent->lchild = temp;
    return  root;
}

//Inorder Traversal - For printing binary search tree - Should be in Ascending Order 
void inorder(struct node *ptr)
{
  if(ptr == NULL)
    return;
  inorder(ptr->lchild);
  cout<<ptr->info<<" ";
  inorder(ptr->rchild);
}

//Recursive Function for calculating Height of Binary Search Tree
int height (struct node *ptr)
{
  if(ptr == NULL)
    return 0;
  return 1+max(height(ptr->lchild),height(ptr->rchild));
}

int main()
{
   int t ;     //t is number of nodes
   cout<<"Enter number of nodes "<<endl;
   cin>>t;   
   struct node *root = NULL ;
   cout<<"Enter values of nodes"<<endl;
   while(t--)
   {
     int n;     //n is value of node
     cin>>n;
     root = insert_bs(root , n);
   }

   //For printing constructed binary Tree uncomment "inorder" function
   // inorder(root);    
   // cout<<endl;
   
   cout<<"Height of given Binary Search Tree is "<<height(root)<<endl;
}

File: Algorithms/Tree/Height of the tree/Java/bst_height.java
import java.util.Scanner;
import java.lang.*; 
// Finding height of a binary search tree ;

// Node of a tree
class Node{
    int data ;
    Node left , right ;

    Node(int val){
        data = val;
        left = null ;
        right = null ;
    }
}

public class Tree{
    Node root ;
    Tree(){
        root = null;
    }

    // Function to insert node in BST 
    void insert(Node r , int key){
        root= insertHelper(r ,key);
    }
    Node insertHelper(Node r, int key) { 
    
            //If the tree is empty, create a new node and return
            if (r == null) { 
                r = new Node(key); 
                return r; 
            } 
            
            // else traverse down.
            if (key < r.data) 
                r.left = insertHelper(r.left, key); 
            else if (key > r.data) 
                r.right = insertHelper(r.right, key); 
    
            return r;
    }
    // FUNCTION 
    //Recursive Function for calculating Height of Binary Search Tree
    int height(Node node){
        if(node == null){
            return 0; 
        }
        else{
            return 1 + Math.max(height(node.left), height(node.right));
        }
    }
    
    
    void inorder()  { 
       inorderHelper(root); 
    } 
  
    // Function to do inorder traversal of BST 
    void inorderHelper(Node root) { 
        if (root != null) { 
            inorderHelper(root.left); 
            System.out.println(root.data); 
            inorderHelper(root.right); 
        } 
    } 

    // main function to test the height function
    public static void main(String args[]){
        Scanner in = new Scanner(System.in);
        int test = in.nextInt(); //test is number of nodes 
        Tree tree = new Tree();
        for( ; test > 0 ; test--){
            int temp = (int)(Math.random()*100);
            tree.insert(tree.root,temp); // Inserting random numbers in tree ;
        }
        
        //For printing constructed binary Tree uncomment the line below
        //tree.inorder(); 
  
        System.out.println("Height of the tree is " + tree.height(tree.root));
    }
}